src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotOverloadOperatorEqualsOnReferenceTypesTests.cs

```diff
@@ -0,0 +1,264 @@
+        }
+
+        [Fact]
+        public async Task OperatorEqualAndAdditionSubtraction_ReferenceType_Diagnostic()
+        {
+            // Doc states that if type behaves as a value-type and has addition/subtraction it might be safe.
+
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public class C
+{
+    public static bool operator [|==|](C left, C right) => true;
+    public static bool operator !=(C left, C right) => true;
+
+    public static C operator +(C left, C right) => left;
+    public static C operator -(C left, C right) => left;
+}");
+
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Public Class C
+    Public Shared Operator [|=|](ByVal left As C, ByVal right As C) As Boolean
+        Return True
+    End Operator
+
+    Public Shared Operator <>(ByVal left As C, ByVal right As C) As Boolean
+        Return True
+    End Operator
+
+    Public Shared Operator +(ByVal left As C, ByVal right As C) As C
+        Return left
+    End Operator
+
+    Public Shared Operator -(ByVal left As C, ByVal right As C) As C
+        Return left
+    End Operator
+End Class");
+        }
+
+        [Theory]
+        // General analyzer option
+        [InlineData("public", "dotnet_code_quality.api_surface = public")]
+        [InlineData("public", "dotnet_code_quality.api_surface = private, internal, public")]
+        [InlineData("public", "dotnet_code_quality.api_surface = all")]
+        [InlineData("protected", "dotnet_code_quality.api_surface = public")]
+        [InlineData("protected", "dotnet_code_quality.api_surface = private, internal, public")]
+        [InlineData("protected", "dotnet_code_quality.api_surface = all")]
+        [InlineData("internal", "dotnet_code_quality.api_surface = internal")]
+        [InlineData("internal", "dotnet_code_quality.api_surface = private, internal")]
+        [InlineData("internal", "dotnet_code_quality.api_surface = all")]
+        [InlineData("private", "dotnet_code_quality.api_surface = private")]
+        [InlineData("private", "dotnet_code_quality.api_surface = private, public")]
+        [InlineData("private", "dotnet_code_quality.api_surface = all")]
+        // Specific analyzer option
+        [InlineData("internal", "dotnet_code_quality.CA1046.api_surface = all")]
+        [InlineData("internal", "dotnet_code_quality.Design.api_surface = all")]
+        // General + Specific analyzer option
+        [InlineData("internal", @"dotnet_code_quality.api_surface = private
+                                  dotnet_code_quality.CA1046.api_surface = all")]
+        // Case-insensitive analyzer option
+        [InlineData("internal", "DOTNET_code_quality.CA1046.API_SURFACE = ALL")]
+        // Invalid analyzer option ignored
+        [InlineData("internal", @"dotnet_code_quality.api_surface = all
+                                  dotnet_code_quality.CA1046.api_surface_2 = private")]
+        public async Task CSharp_ApiSurfaceOption(string accessibility, string editorConfigText)
```

Comment:
For future reference, I don’t think we need such a huge matrix of tests for api_surface option. Just couple of sanity checks for default behavior (fires for public and not for internal) and configuration (can be made to fire for internal with editorconfig) should suffice.

Reply:
Ok thank you for the info! That will save me some keystrokes!